support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerListener.java

```diff
@@ -29,4 +35,24 @@ public void onDeleted(Run<?,?> run) {
         }
     }
 
+    @Extension
+    public static final class Listener extends RunListener<Run<?, ?>> {
+        @Override
+        public void onStarted(Run<?, ?> run, TaskListener listener) {
+            Cause.UpstreamCause upstreamCause = run.getCause(Cause.UpstreamCause.class);
+            if (upstreamCause != null) {
+                Run upStreamRun = upstreamCause.getUpstreamRun();
+                BuildTriggerAction buildTriggerAction = run.getAction(BuildTriggerAction.class);
+                //Only works when upstream is still running
+                if (upStreamRun != null && upStreamRun.isBuilding() && buildTriggerAction != null) {
```

Comment:
I do not think you need to bother checking for `upstreamCause`/`upStreamRun` at all. If the upstream build has already finished, the `TaskListener` from its `StepContext` should just be ignored. You can call `StepContext.isReady` I suppose.